### PR TITLE
adding subGoalCheckField

### PR DESCRIPTION
**Pull request by JanuszJakubiec**

### CODE REVIEW

File: planner-api/src/test/java/pl/edu/agh/kuce/planner/goal/service/GoalServiceWithSubGoals.java [hunk 1]
```diff
@@ -121,4 +122,60 @@ void checkIfProperGoalIsReturnedAfterSubGoalCreation() {
         goalList = goalService.list(user);
         assertThat(goalList.goals().get(0).amount()).isEqualTo(777);
     }
+
+    @Test
+    void checkIfSubGoalsAreInCompleteOnTheResponseList() {
+        final User user = userRepository.save(new User("TEST28", "TEST28", "TEST28"));
+        final Goal goal = goalRepository.save(new Goal(user, "TEST4", 28));
+        subGoalRepository.save(new SubGoal(goal, "TEST", 322));
```
Comment (marszub) [inline planner-api/src/test/java/pl/edu/agh/kuce/planner/goal/service/GoalServiceWithSubGoals.java:130]: If you test subgoal service, you should create them with service

File: planner-api/src/main/java/pl/edu/agh/kuce/planner/goal/service/GoalService.java [hunk 2]
```diff
@@ -127,4 +126,17 @@ private List<SubGoalData> transformSubGoals(final List<SubGoal> subGoalList) {
...
                           .toList();
     }
+
+    @Transactional
+    public GoalData completeSubGoal(final Integer subGoalId, final Integer goalId, final User user) {
+        final Optional<Goal> goal = goalRepository.getGoalById(goalId, user);
+        if (goal.isPresent()) {
+            final Optional<SubGoal> subGoal = subGoalRepository.getSubGoalById(subGoalId, user);
+            if (subGoal.isPresent()) {
+                subGoalRepository.completeSubGoal(subGoalId, goal.get());
```
Comment (marszub) [inline planner-api/src/main/java/pl/edu/agh/kuce/planner/goal/service/GoalService.java:136]: All the methot besides this line is the same as destroySubGoal. Consider extracting a new method. In addidtion, method destroy also has duplicate part (maybe 2 new methods?)

File: planner-api/src/main/java/pl/edu/agh/kuce/planner/goal/persistence/SubGoal.java [hunk 3]
```diff
@@ -31,20 +31,32 @@ public class SubGoal {
...
+        this.completed = Boolean.FALSE;
     }
 
     public SubGoal(final Goal goal, final SubGoalInputData data) {
         this(
                 goal,
                 data.title(),
-                data.amount()
+                data.amount(),
+                Boolean.FALSE
```
Comment (marszub) [inline planner-api/src/main/java/pl/edu/agh/kuce/planner/goal/persistence/SubGoal.java:59]: Consider using SubGoal(final Goal goal, final String title, final Integer amount) constructor instead. Then You could delete SubGoal(final Goal goal, final String title, final Integer amount, final Boolean completed).

File: planner-api/src/main/java/pl/edu/agh/kuce/planner/goal/persistence/SubGoal.java [hunk 4]
```diff
@@ -31,20 +31,32 @@ public class SubGoal {
     @Column(nullable = false)
     private Integer amount;
 
+    @Column(nullable = false)
+    private Boolean completed;
+
     public SubGoal() {
```
Comment (marszub) [inline planner-api/src/main/java/pl/edu/agh/kuce/planner/goal/persistence/SubGoal.java:37]: This constructor is not used. Could be deleted

File: planner-api/src/main/java/pl/edu/agh/kuce/planner/goal/GoalController.java [hunk 5]
```diff
@@ -68,4 +69,11 @@ public GoalData destroy(@PathVariable("id") final Integer goalId,
                         @PathVariable("subGoalId") final Integer subGoalId) {
         return goalService.destroySubGoal(subGoalId, goalId, user);
     }
+
+    @PutMapping("/{id}/sub-goals/{subGoalId}")
```
Comment (marszub) [inline planner-api/src/main/java/pl/edu/agh/kuce/planner/goal/GoalController.java:73]: You update only part of the resource, so patch would be more accurate

File: planner-api/openapi.yaml [hunk 6]
```diff
@@ -309,6 +309,19 @@ paths:
           description: "User unauthorized"
         "404":
           description: "SubGoal with that ID does not exist"
+    put:
```
Comment (marszub) [inline planner-api/openapi.yaml:312]: You update only part of the resource, so patch would be more accurate

File: planner-api/openapi.yaml [hunk 7]
```diff
@@ -572,6 +585,12 @@ components:
                     minLength: 1
                     maxLength: 200
                     example: "Winter tyres"
+                  amount:
```
Comment (marszub) [inline planner-api/openapi.yaml:588]: After these changes, GoalData became identical as the objects in this array. You could replace the type definition with reference to GoalData scheme

**Review by marszub — CHANGES_REQUESTED** [review]
Put to patch
Api doc scheme reference in place of full object definition
Redundant constructors

File: planner-api/src/main/java/pl/edu/agh/kuce/planner/goal/persistence/SubGoal.java (hunk 4 above)
Comment (JanuszJakubiec) [reply planner-api/src/main/java/pl/edu/agh/kuce/planner/goal/persistence/SubGoal.java:37]: Unfortunatelly It can not be deleted :(

Comment (JanuszJakubiec) [reply planner-api/src/main/java/pl/edu/agh/kuce/planner/goal/persistence/SubGoal.java:37]: There are errors that SubGoal does not have default constructor

**Review by marszub — APPROVED** [review]

**Pull request merged**